hyper/http11/connection.py

```diff
@@ -8,15 +8,18 @@
 import logging
 import os
 import socket
+import base64
 
 from .response import HTTP11Response
 from ..tls import wrap_socket
 from ..common.bufsocket import BufferedSocket
-from ..common.exceptions import TLSUpgrade
+from ..common.exceptions import TLSUpgrade, HTTPUpgrade
 from ..common.headers import HTTPHeaderMap
 from ..common.util import to_bytestring
 from ..compat import bytes
 
+from ..http20.connection import H2C_PROTOCOL
```

Comment:
It'd be better not to have an import from `http20` if at all possible. Can we pull the H2C_PROTOCOL declaration out into common code?

It might be that the best place for it is actually `tis`.